pkg/p2p/libp2p/libp2p.go

```diff
@@ -200,6 +201,13 @@ func New(ctx context.Context, o Options) (*Service, error) {
 	return s, nil
 }
 
+func (s *Service) AddService(service ProtocolService) {
```

Comment:
if we modify `p2p.ProtocolSpec` to the following:

```go
type ProtocolSpec struct {
	Name        string
        Init             func() error
	StreamSpecs []StreamSpec
}
```
this function could be avoided, and the init handler could be specified directly alongside the protocol specification. but also clarity is always subjective so i'd be happy if others express their opinions on this too

Reply:
I agree, good suggestion. This approach is in line with specifying stream Handlers in ProtocolSpec in declarative way. The same goes for the future peer Close (OnDisconnect) function for protocols.

Reply:
Yup, I agree, I left it "dummy" way to hear you guys suggestion for this. I like it, thanks!